### ISSUE TITLE
yt-proxy и иже с ним

### ISSUE THREAD

**Issue by Baltazar500** [mesb1/xupnpd_youtube#1]
yt-dlp опять гавкнулся и удалённый прокси надо обновлять. Решил открыть issue в вашей ветке xupnpd_youtube, чтобы лишний раз ветку с issue по yt-proxy не захламлять

Самое забавное, что на такой случай я проработал альтернативный вариант с получением линка через внешний сервис, вот только всё на корню сейчас РКН зарубил со своим замедлением. Линк начинает играться, но буквально сразу же дохнет т.к. скорость передачи данных идёт на 128 - 384 кбит/с (16 - 32 кб/с), а этого не хватает даже для 360p. С одной стороны обложил сам ютуб, с другой РКН. Молодцы, чо. Ппц.

**Comment by mesb1** [comment]
Обновлено, проверяем.

**Comment by Baltazar500** [comment]
Заработало :) Вот только новые линки работают нормально, а старые валятся в corrupted ибо закешировались в памяти. Надо сам проксик ещё перезапустить.

**Comment by mesb1** [comment]
Готово.
Но в целом если что то там линки имеют какой-то таймаут - и он учитываяется проксиком, просто в какойто момент он истечёт и проксик снова переуточнит данные перезапустив длп.

**Comment by Baltazar500** [comment]
>Готово.

Работает :)

>Но в целом если что то там линки имеют какой-то таймаут - и он учитываяется проксиком, просто в какойто момент он истечёт и проксик снова переуточнит данные перезапустив длп.

Да это я помню. Просто там по факту были не валидные линки (ошибка 403), которым висеть в памяти ещё как минимум несколько часов (навскидку сейчас не вспомню сколько конкретно), потому и попросил :)

**Comment by Baltazar500** [comment]
Гугл озверел. Опять сломал yt-dlp и проксик. Апдейт ещё не выкатили, хотя свежая ночнушка (не проверял) была сделана 18 часов назад. Но тогда ещё всё было ОК. Так что походу ждём.

~~upd: Походу 18-й формат (360p) грохнули. Приехали ...~~

upd: "nsig extraction failed: Some formats may be missing", ждём корректировок

**Comment by mesb1** [comment]
Непонял что за проблема то.
Несмотря на ошибки 18 формат там в наличии же.

**Comment by mesb1** [comment]
На всякий подвинул версию на мастер покамесь.

**Comment by Baltazar500** [comment]
>Непонял что за проблема то.

Изменился то ли блок расшифровки ссылок криптованного блока из ютубовмкого api, то ли параметра "n" в ссылках (не сильно вникал), из-за чего ряд форматов, включая 360p отвалился на тот момент.

**Comment by mesb1** [comment]
С мастера переместился на стейбл от первого августа если что.
Наблюдаем.

**Comment by Baltazar500** [comment]
Опять гугл всё сломал. Это ппц какой-то. Самое забавное, что внешний сервис сейчас отдаёт корректный линк (или алгоритм другой используют или уже пофиксили у себя), но скорость в ноль зарезает ТСПУ/DPI в РФ. В общем, со всех сторон обложили. Начинаю подумывать об освоении vps. Ранее опыта не было, но думаю, разберусь. Есть что на примете ? Адекватное в плане цены и оплаты из РФ ? VPS в РФ ? VPS за бугром ?

**Comment by mesb1** [comment]
Часто вижу в рекомендациях aeza но ябы скорее ориентировался на доступность нужных ресурсов и цену.
Такчто взяли попробовать - погоняли на скорость\доступность и дальше сидим.

**Comment by mesb1** [comment]
Попутно обновился до 20240806.

**Comment by Baltazar500** [comment]
>Часто вижу в рекомендациях aeza но ябы скорее ориентировался на доступность нужных ресурсов и цену. Такчто взяли попробовать - погоняли на скорость\доступность и дальше сидим.

ОК. Буду посмотреть.Спасибо. Я в этом деле пока что профан и хз с чего начинать. Начну с aez'ы, а там будет видно ...

>Попутно обновился до 20240806

Мб имеет смысл поставить на автоапдейт в cron ? Или чревато ? Просто в нынешних условиях, когда гугл раз в неделю что-то херит это имеет смысл.

**Comment by mesb1** [comment]
> Мб имеет смысл поставить на автоапдейт в cron ? Или чревато ? Просто в нынешних условиях, когда гугл

Посмотрим, если меня это будет доставать то попробую.
У такого типа хостов я не очень хочу возиться с погружением в настройки.
Учитывая обстановку, сегодня он нужен, завтра менять. Больше ковыряний чем скопировать папку и запуск нужного ябы не хотел.
Так у меня с вдсиной случилось например, настоятельно не рекомендую)

**Comment by mesb1** [comment]
Так, этот хост покамесь нагрузил ещё одной задачей по стримингу - если вдруг обнаружите какие-то сложности в работе маякните, унесу наверное куда-то в другое место.

**Comment by Baltazar500** [comment]
>Так у меня с вдсиной случилось например, настоятельно не рекомендую)

Судя по всему, к ней вообще лучше не соваться. Во многих местах в её сторону пишут не лестные отзывы по самым разным причинам.

>Так, этот хост покамесь нагрузил ещё одной задачей по стримингу - если вдруг обнаружите какие-то сложности в работе маякните, унесу наверное куда-то в другое место.

Вроде особых проблем не видел, но сейчас походу как обычно поломался yt-dlp. Ролики опять в "corrupted" валятся.

**Comment by mesb1** [comment]
Пока непонятно.
Ишью канеш на гитхабе у них есть.
Пока на моих хостах я получаю при попытках листинга форматов сразу:
Sign in to confirm you’re not a bot. This helps protect our community. Learn more

**Comment by Baltazar500** [comment]
>Sign in to confirm you’re not a bot. This helps protect our community. Learn more

Ловил у себя на десктопе такое. Причём 1-й запрос нормально проходит, а дальше - отлуп. Какой-то автобан срабатывает. На другом ip-адресе отдавало линки без заморочек при этом. Мб там как-то линки через связку с  pahantom-js получать, эмулируя запросы браузера ? Надо справки навести. Что интересно, подобную петрушку наблюдаю и на серверах проекта pipedapi.

**Comment by Baltazar500** [comment]
Так. А что если yt-dlp авторизационные куки от учётки гуглоакка подкинуть ? Если есть учётка, которую нежалко - можно попробовать. Если не прокатит, можно попробовать варинт с парсингом api фронтенда invidious. Как руки дойдут запилю скрипт. Жаль, что про эти фронтенды узнал только сейчас. Раньше бы они были неплохим подспорьем при получении ссылок локально. Сейчас pipedapi с его кэшированными/проксированными линками практически сломан из-за гугловской борьбы, а invidious локкнут РКН (на целевом плеере со средствами обхода никак от слова "совсем") .

**Comment by mesb1** [comment]
Ну вообще я пролистал это вот:

https://github.com/yt-dlp/yt-dlp/issues/10128

Местами пишут про бан на 72 часа.
Местами пишут про oauth.

Сходу так и не понял умеет ли дефолтная длп в последнее или надо юзать какие-то форки.

**Comment by mesb1** [comment]
Вобщем раскопал свой старый акк от ютуба.
И с помощью nightly dlp и проекта 
https://github.com/coletdjnz/yt-dlp-youtube-oauth2

Приаттачился к этому старому акку.
Вобщем вроде как заколосилось - проверяем насколько этого хватит.

**Comment by Baltazar500** [comment]
>Вобщем вроде как заколосилось - проверяем насколько этого хватит.

Если долго не проживёт, скину скрипт тягающий ссылки у фронтенда invidious. Проку у меня от него никакого, а на удалённом сервере с отсутствием блока РКН в нём есть смысл :)

**Comment by mesb1** [comment]
> Проку у меня от него никакого, а на удалённом сервере с отсутствием блока РКН в нём есть смысл :)

Прикол кстати что блок ркн есть, только он распространяется на впн протоколы)
А эту штуку никто не трогает как и ссх доступ.
